src/windows_file_codecvt.cpp

```diff
@@ -59,8 +60,10 @@
     UINT codepage = AreFileApisANSI() ? CP_ACP : CP_OEMCP;
 
     int count;
+    // Use static cast to silence Visual Studio 2017:
+    // warning C4244: 'argument': conversion from '__int64' to 'int', possible loss of data
     if ((count = ::WideCharToMultiByte(codepage, WC_NO_BEST_FIT_CHARS, from,
-      from_end - from, to, to_end - to, 0, 0)) == 0)
+       boost::numeric_cast<int>(from_end - from), to, boost::numeric_cast<int>(to_end - to), 0, 0)) == 0)
```

Comment:
Sorry for nitpicking again. This line has 1 extra indentation space, compared to the original line.

Also, this line has become very long. I'd rewrite the whole fragment, for example, as:

```c++
int count = ::WideCharToMultiByte(
  codepage, WC_NO_BEST_FIT_CHARS,
  from, boost::numeric_cast<int>(from_end - from),
  to, boost::numeric_cast<int>(to_end - to),
  0, 0);
if (count == 0)
  return error;  // conversion failed
```